### ISSUE TITLE
Error in vapply(otus, function(taxa) { : values must be type 'integer',  but FUN(X[[1]]) result is type 'double'

### ISSUE THREAD

**Issue by Debora6991** [microbiome/microbiome#148]
Hi there,
I'm trying to follow step by step a microbiome tutorial (https://microsud.github.io/microbiometutorials.gihub.io/microbiome_tutorial.html#load-the-libraries) and I can't manage to solve a few issues.

-the first one concerns the divergence function, that I had already succesfully used for other analyses, getting this error:

Error in as.vector(y) : argument "y" is missing, with no default

-the second issue is with the "aggregate_top_taxa" function (and the aggregate_taxa one), where I get this error:

Error in vapply(otus, function(taxa) { : values must be type 'integer', but FUN(X[[1]]) result is type 'double'
I was able to use that in the past too.

I asked a colleague to run the same script on my data in her own computer and she managed to make both the functions work.
The only difference I can think of is the R version, mine is 4.0.3 and hers is 3.6.3. I downloaded ex novo all packages and updated everything but still I cant manage to mate them work.

Any Idea why is this  happening?

Thanks in advance for your help,

Debora

**Comment by microsud** [comment]
hi @Debora6991 
Thanks for raising the query. The tutorials you are working with are from 2017 and not updated. Parts of it are moved to the main package tutorials at this [weblink](https://microbiome.github.io/tutorials/). I will remove the old ones. Apologies for inconvenience.  

Regarding the `aggregate_top_taxa` this function is now not supported and an alternative is available `aggregate_rare`.
Please check the latest updates mentioned in the [news](https://github.com/microbiome/microbiome/blob/master/NEWS)

If the issue persists please report it here. I will keep this issue open. 

Best wishes,
Sudarshan

**Comment by Debora6991** [comment]
I followed the tutorial you mentioned but nothing seems to change, I still
get the two same errors.
I can't figure out what the problem is here.

-not the data, it works on another person's computer.
-not the microbiome version, I reinstalled it

Could the latest R version be the problem?

Il giorno ven 20 nov 2020 alle ore 11:26 Sudarshan <notifications@github.com>
ha scritto:

**Comment by gaferguz** [comment]
Hi @microsud and @Debora6991 , I experienced the same error when using aggregate_rare or aggregate_taxa functions on my PhyloSeq object in Rstudio - R version 4.0.2 (2020-06-22). It only occurs when I run it in Mac Big Sur. I have tried on the same object in Ubuntu 18.04.5 lts and works perfectly fine.

``
x <- microbiome::aggregate_taxa(Phylo, level = "Phylum") 
``
``
Error in vapply(otus, function(taxa) { : values must be type 'integer',but FUN(X[[1]]) result is type 'double'
``
